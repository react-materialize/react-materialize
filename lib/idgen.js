"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.resetID = resetID;
exports.default = idgen;
var id = 0;

function resetID() {
  id = 0;
}

function idgen() {
  var oldId = id;
  id += 1;
  return oldId;
}