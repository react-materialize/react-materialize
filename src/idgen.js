let id = 0

export function resetID() {
  id = 0
}

export default function idgen() {
  let oldId = id
  id += 1
  return oldId
}
