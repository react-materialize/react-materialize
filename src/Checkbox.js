import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import idgen from './idgen';

class Checkbox extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const { indeterminate } = this.props;

    if (this._input && indeterminate) {
      this._input.indeterminate = indeterminate;
      this._input.checked = false;
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.indeterminate !== this.props.indeterminate) {
      this._input.indeterminate = this.props.indeterminate;
    }
  }

  render() {
    const {
      id,
      className,
      indeterminate,
      filledIn,
      disabled,
      onChange,
      checked,
      value,
      label
    } = this.props;

    let computedId = id || idgen();

    if (indeterminate) {
      computedId = `indeterminate-checkbox-${idgen()}`;
    }

    return (
      <label htmlFor={computedId}>
        <input
          id={computedId}
          className={cx(
            {
              'filled-in': filledIn
            },
            className
          )}
          disabled={disabled}
          onChange={onChange}
          type="checkbox"
          defaultChecked={checked}
          value={value}
          ref={input => (this._input = input)}
        />
        <span>{label}</span>
      </label>
    );
  }
}

Checkbox.propTypes = {
  className: PropTypes.string,
  /*
   * checkbox value
   */
  value: PropTypes.string.isRequired,
  /*
   * filled-in styled checkbox
   */
  filledIn: PropTypes.bool,
  /*
   * label next to checkbox
   */
  label: PropTypes.string.isRequired,
  /*
   * Indeterminate Style
   */
  indeterminate: PropTypes.bool,
  /*
   * onChange callback
   */
  onChange: PropTypes.func,
  /*
   * override id
   * @default idgen()
   */
  id: PropTypes.string,
  /*
   * disabled input
   */
  disabled: PropTypes.bool,
  /*
   * A Boolean attribute indicating whether or not this checkbox is checked
   */
  checked: PropTypes.bool
};

export default Checkbox;
