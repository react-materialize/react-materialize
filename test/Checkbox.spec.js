import React from 'react';
import { shallow, mount } from 'enzyme';
import Checkbox from '../src/Checkbox';
import idgen from '../src/idgen';

jest.mock('../src/idgen');

describe('<Checkbox />', () => {
  let wrapper;

  beforeEach(() => {
    idgen.mockClear();
  });

  test('renders', () => {
    wrapper = shallow(<Checkbox value="red" label="red" />);
    expect(idgen).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders disabled', () => {
    wrapper = shallow(<Checkbox value="red" label="red" disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  test('filled-in', () => {
    wrapper = shallow(<Checkbox value="red" label="red" filledIn />);
    expect(wrapper).toMatchSnapshot();
  });

  test('with id', () => {
    wrapper = shallow(<Checkbox value="red" label="red" id="test" />);
    expect(idgen).not.toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  test('with indeterminate', () => {
    wrapper = mount(<Checkbox value="red" label="red" indeterminate />);
    expect(wrapper.find('input').instance().checked).toBe(false);
    expect(wrapper.find('input').instance().indeterminate).toBe(true);
  });

  test('checked', () => {
    wrapper = mount(<Checkbox value="red" label="red" checked />);
    expect(wrapper.find('input').instance().checked).toBe(true);
    expect(wrapper.find('input').instance().indeterminate).toBe(false);
  });

  test('does not overwrite materialize-css classnames', () => {
    wrapper = mount(
      <Checkbox value="red" label="red" filledIn className="my-class" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
