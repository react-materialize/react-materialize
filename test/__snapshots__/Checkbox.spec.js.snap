// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> does not overwrite materialize-css classnames 1`] = `
<Checkbox
  className="my-class"
  filledIn={true}
  label="red"
  value="red"
>
  <label
    htmlFor="mockId"
  >
    <input
      className="filled-in my-class"
      id="mockId"
      type="checkbox"
      value="red"
    />
    <span>
      red
    </span>
  </label>
</Checkbox>
`;

exports[`<Checkbox /> filled-in 1`] = `
<label
  htmlFor="mockId"
>
  <input
    className="filled-in"
    id="mockId"
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> renders 1`] = `
<label
  htmlFor="mockId"
>
  <input
    className=""
    id="mockId"
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> renders disabled 1`] = `
<label
  htmlFor="mockId"
>
  <input
    className=""
    disabled={true}
    id="mockId"
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> with id 1`] = `
<label
  htmlFor="test"
>
  <input
    className=""
    id="test"
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;
